src/pages/index.js

```diff
@@ -0,0 +1,348 @@
+        openModal(previewModal);
+        previewModalImageEl.src = data.link;
+        previewModalCaptionEl.textContent = data.name;
+        previewModalImageEl.alt = data.name;
+    });
+
+    cardDeleteBtn.addEventListener("click", () => {
+        handleDeleteCard(cardElement, data._id);
+    });
+
+    return cardElement;
+}
+
+function openModal(modal) {
+    modal.classList.add("modal_opened");
+    document.addEventListener("keyup", handleEscape);
+}
+
+function closeModal(modal) {
+    modal.classList.remove("modal_opened");
+    document.removeEventListener("keyup", handleEscape);
+}
+
+function handleEditFormSubmit(evt) {
+    evt.preventDefault();
+
+    const submitBtn = evt.submitter;
+    setbuttonText(submitBtn, true, "Saving...", "Save");
+
+    api.editUserInfo({
+        name: editModalNameInput.value,
+        about: editModalDescriptionInput.value,
+    })
+        .then((data) => {
+            profileName.textContent = data.name;
+            profileDescription.textContent = data.about;
+            closeModal(editProfileModal);
+        })
+        .catch(console.error)
+        .finally(() => {
+            setbuttonText(submitBtn, false, "Saving...", "Save");
+        });
+}
+
+function handleAddCardSubmit(evt) {
+    evt.preventDefault();
+
+    const inputValues = {
+        name: cardNameInput.value,
+        link: cardLinkInput.value,
+    };
+
+    api.addCard(inputValues)
+        .then((cardData) => {
+            const cardElement = getCardElement(cardData);
+            cardsList.prepend(cardElement);
+            evt.target.reset();
+            disabledButton(cardSubmitBtn, settings);
+            closeModal(cardModal);
+        })
+        .catch(console.error);
+}
+
+function handleAvatarSubmit(evt) {
+    evt.preventDefault();
+
+    const submitBtn = evt.submitter || avatarSubmitBtn;
+    setbuttonText(submitBtn, true, "Saving...", "Save");
+
+    const avatarUrl = avatarInput.value;
+    if (!avatarUrl) {
+        console.error("Avatar URL is empty.");
+        setbuttonText(submitBtn, false, "Saving...", "Save");
+        return;
+    }
+
+    api.editAvatarInfo(avatarUrl)
+        .then((data) => {
+            if (data && data.avatar) {
+                const profileAvatar =
+                    document.querySelector(".profile__avatar");
+                if (profileAvatar) {
+                    profileAvatar.src = data.avatar;
+                } else {
+                    console.error("Profile avatar element not found.");
+                }
+                closeModal(avatarModal);
+                evt.target.reset();
+            } else {
+                console.error("API did not return a valid avatar URL.", data);
+            }
+        })
+        .catch(console.error)
+        .finally(() => {
+            setbuttonText(submitBtn, false, "Saving...", "Save");
+        });
+}
+
+function handleDeleteSubmit(evt) {
+    evt.preventDefault();
+    api.deleteCard(selectedCardId)
```

Comment:
According to the figma design, the delete modal button should show `Deleting...` while deleting a card